rdkit/Chem/AtomPairs/Torsions.py

```diff
@@ -93,71 +87,75 @@ def ExplainPathScore(score, size=4):
-  >>> ExplainPathScore(score,3)
+  >>> score=pyScorePath(m, (0, 1, 2), 3)
+  >>> ExplainPathScore(score, 3)
   (('C', 1, 0), ('C', 2, 1), ('C', 1, 1))
 
   Again, it's order independent:
 
-  >>> score=pyScorePath(m,(2,1,0),3)
-  >>> ExplainPathScore(score,3)
+  >>> score=pyScorePath(m, (2, 1, 0), 3)
+  >>> ExplainPathScore(score, 3)
   (('C', 1, 0), ('C', 2, 1), ('C', 1, 1))
 
   >>> m = Chem.MolFromSmiles('C=CO')
-  >>> score=pyScorePath(m,(0,1,2),3)
-  >>> ExplainPathScore(score,3)
+  >>> score=pyScorePath(m, (0, 1, 2), 3)
+  >>> ExplainPathScore(score, 3)
   (('C', 1, 1), ('C', 2, 1), ('O', 1, 0))
 
   >>> m = Chem.MolFromSmiles('OC=CO')
-  >>> score=pyScorePath(m,(0,1,2,3),4)
-  >>> ExplainPathScore(score,4)
+  >>> score=pyScorePath(m, (0, 1, 2, 3), 4)
+  >>> ExplainPathScore(score, 4)
   (('O', 1, 0), ('C', 2, 1), ('C', 2, 1), ('O', 1, 0))
 
   >>> m = Chem.MolFromSmiles('CC=CO')
-  >>> score=pyScorePath(m,(0,1,2,3),4)
-  >>> ExplainPathScore(score,4)
+  >>> score=pyScorePath(m, (0, 1, 2, 3), 4)
+  >>> ExplainPathScore(score, 4)
   (('C', 1, 0), ('C', 2, 1), ('C', 2, 1), ('O', 1, 0))
 
 
   >>> m = Chem.MolFromSmiles('C=CC(=O)O')
-  >>> score=pyScorePath(m,(0,1,2,3),4)
-  >>> ExplainPathScore(score,4)
+  >>> score=pyScorePath(m, (0, 1, 2, 3), 4)
+  >>> ExplainPathScore(score, 4)
   (('C', 1, 1), ('C', 2, 1), ('C', 3, 1), ('O', 1, 1))
-  >>> score=pyScorePath(m,(0,1,2,4),4)
-  >>> ExplainPathScore(score,4)
+  >>> score=pyScorePath(m, (0, 1, 2, 4), 4)
+  >>> ExplainPathScore(score, 4)
   (('C', 1, 1), ('C', 2, 1), ('C', 3, 1), ('O', 1, 0))
 
 
   >>> m = Chem.MolFromSmiles('OOOO')
-  >>> score=pyScorePath(m,(0,1,2),3)
-  >>> ExplainPathScore(score,3)
+  >>> score=pyScorePath(m, (0, 1, 2), 3)
+  >>> ExplainPathScore(score, 3)
   (('O', 1, 0), ('O', 2, 0), ('O', 2, 0))
-  >>> score=pyScorePath(m,(0,1,2,3),4)
-  >>> ExplainPathScore(score,4)
+  >>> score=pyScorePath(m, (0, 1, 2, 3), 4)
+  >>> ExplainPathScore(score, 4)
   (('O', 1, 0), ('O', 2, 0), ('O', 2, 0), ('O', 1, 0))
 
   """
   codeMask = (1 << rdMolDescriptors.AtomPairsParameters.codeSize) - 1
   res = [None] * size
   for i in range(size):
-    if i == 0 or i == (size - 1):
-      sub = 1
-    else:
-      sub = 2
+    sub = 1 if i in (0, size - 1) else 2
     code = score & codeMask
     score = score >> rdMolDescriptors.AtomPairsParameters.codeSize
     symb, nBranch, nPi = Utils.ExplainAtomCode(code)
-    expl = symb, nBranch + sub, nPi
-    res[i] = expl
+    res[i] = (symb, nBranch + sub, nPi)
   return tuple(res)
 
 
 def GetTopologicalTorsionFingerprintAsIds(mol, targetSize=4):
+  """
+  Old code: (Clean if merged)
   iv = GetTopologicalTorsionFingerprint(mol, targetSize)
   res = []
   for k, v in iv.GetNonzeroElements().items():
     res.extend([k] * v)
   res.sort()
+  """
+  nonZeroElements = GetTopologicalTorsionFingerprint(mol, targetSize).GetNonzeroElements()
+  resDict = sorted(nonZeroElements.items())
+  res = []
+  for k, v in resDict:
+    res.extend([k] * v)
```

Comment:
The name `resDict` gives the impression that we have a dictionary. It's however a list of tuples. You could call it `frequencies` instead.